### ISSUE TITLE
`W0631:undefined-loop-variable` false positive when for loop has else branch with `break` or `continue

### ISSUE THREAD

**Issue by DetachHead** [PyCQA/pylint#7311]
### Bug description

```py
def foo(l):
    while True:
        for i in l:
            break
        else:
            continue
            # or break
        print(i)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:8:14: W0631: Using possibly undefined loop variable 'i' (undefined-loop-variable)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
```

**Comment by jacobtylerwalls** [comment]
I'd be in favor of adding `break` and `continue` to the list of nodes checked in the PR for #6480.

Would you be interested to prepare a PR?

**Comment by DanielNoord** [comment]
@DetachHead Do you want to work on the `NoReturn` example? I can guide you through it if you want.

If you get the `FunctionDef` node of `fail` by inferring the call to it you can use the `type_comment_returns` attribute to get the return type annotation. With an inference on that annotation + a check for `.qualname() == "typing.NoReturn` I think we should be able to add this case as well.

**Comment by DetachHead** [comment]
@DanielNoord i gave it a go but i can't figure out how to get the `FunctionDef` from the function call. #7419
